packages/@aws-cdk/aws-lambda/lib/layer-version.ts

```diff
@@ -0,0 +1,163 @@
+import cdk = require('@aws-cdk/cdk');
+import { Code, InlineCode } from './code';
+import { Runtime } from './runtime';
+
+export interface LayerVersionProps {
+  /**
+   * The runtimes that this layer is compatible with.
+   *
+   * @default All runtimes are supported
+   */
+  compatibleRuntimes?: Runtime[];
+
+  /**
+   * The content of this Layer. Using the ``InlineCode`` class is not permitted.
+   */
+  code: Code;
+
+  /**
+   * The description the this Lambda Layer.
+   */
+  description?: string;
+
+  /**
+   * The SPDX licence identifier or URL to the license file for this layer.
+   */
+  license?: string;
+
+  /**
+   * The name of the layer.
+   * @default a name will be generated.
+   */
+  name?: string;
+}
+
+/**
+ * A reference to a Lambda Layer version.
+ */
+export abstract class LayerVersionRef extends cdk.Construct {
```

Comment:
Use the more modern style for import/export via an interface ILayerVersion and not a public abstract base class (you could still have an internal base clase if you like)

Reply:
There is no such thing as an `internal base class`- it needs to be exported for JSII if it is extended by an exported type.

Reply:
You are obvsiouly correct, so I would do something like this:

```ts
export interface ILayerVersion { ... }

export abstract class LayerVersionBase implements ILayerVersion { }
export class LayerVersion extends LayerVersionBase { }
class ImportedLayerVersion extends LayerVersionBase { }
```